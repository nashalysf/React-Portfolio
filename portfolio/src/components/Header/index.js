import React, { useEffect } from "react";
const Header = (props) => {
  const {
    categories = [],
    setCurrentCategory,
    currentCategory,
    contactSelected,
    setContactSelected,
  } = props;
  useEffect(() => {
    document.title = (currentCategory.name);
  }, [currentCategory]);

  return (
    <header className="flex-row px-1">
      <a data-testid="link" href="/">
        <h2>Naferi</h2>
      </a>
      <nav className="text-center">
        <ul className="flex-row">
          <li className="mx-2">
            <a data-testid="about" href="#about" onClick={() => setContactSelected(false)}>
              About
            </a>
          </li>
          <li className="mx-2">
            <a href="#resume">Resume</a>
          </li>
          <li className={`mx-2 ${contactSelected && 'navActive'}`}>
            <span onClick={() => setContactSelected(true)}>Contact</span>
          </li>
          {categories.map((category) => (
            <li
              className={`mx-1 ${currentCategory.name === category.name && !contactSelected && `navActive`}`}
              key={category.name}
            >
              <span
                onClick={() => {
                  setCurrentCategory(category);
                  setContactSelected(false);
                }}
              >
                {category.name}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default Header;
