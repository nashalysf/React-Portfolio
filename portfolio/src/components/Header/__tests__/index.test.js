import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';

afterEach(cleanup);

describe("Header component", () => {
  // baseline test
  it('renders', () => {
    render(<Header />);
  });
  // snapshot Test
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('links are visible', () => {
    it('inserts text into the links', () => {
        const { getByTestId } = render(<Header />);
        // eslint-disable-next-line testing-library/prefer-screen-queries
        expect(getByTestId('link')).toHaveTextContent('Naferi');
        // eslint-disable-next-line testing-library/prefer-screen-queries
        expect(getByTestId('about')).toHaveTextContent('About');
    });
  })