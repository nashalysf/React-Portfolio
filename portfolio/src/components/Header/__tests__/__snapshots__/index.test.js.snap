// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="flex-row"
  >
    <a
      data-testid="link"
      href="/"
    >
      <h2>
        Naferi
      </h2>
    </a>
    <nav
      class="text-center"
    >
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
           
          <a
            data-testid="about"
            href="#about"
          >
            About
          </a>
        </li>
        <li
          to="/contact"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-2"
        >
          <a
            href="#resume"
          >
            Resume
          </a>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Projects
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Front
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Back
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
